### PR TITLE
IDEM-56

### PR DESCRIPTION
**Pull request by xiangzhe0812**
format proxy end point
Screen Shot 2022-09-06 at 12 02 46 pm

### CODE REVIEW

File: src/exchange/exchange.service.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
+import { Inject, Injectable, Logger } from "@nestjs/common";
+import { User } from "../data/entities/user.entity";
+import { Request } from "../data/entities/request.entity";
```
Comment (brettvniekerk) [inline src/exchange/exchange.service.ts:3]: Is there not an `index.ts` file for entities that lets you have these all on the same line?

File: src/otp/otp.controller.spec.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
+import { ConfigModule } from "@nestjs/config";
+import { Test, TestingModule } from "@nestjs/testing";
+import { expect } from "chai";
+import { repositoryMockFactory } from "../user/user.service.spec";
+import { KycService } from "../services/KycService";
+import { UserService } from "../user/user.service";
+import { OtpController } from "./otp.controller";
+import { EmailService } from "../services/EmailService";
```
Comment (brettvniekerk) [inline src/otp/otp.controller.spec.ts:8]: Same here for `services`, is there not an `index.ts` file that lets you combine this with line 5?

File: src/user/user.controller.spec.ts [hunk 3]
```diff
@@ -4,9 +4,13 @@ import { UserController } from "./user.controller";
 import { expect } from "chai";
 import { ConfigModule } from "@nestjs/config";
 import { UserService } from "./user.service";
+import { repositoryMockFactory } from "./user.service.spec";
+import { EmailService } from "../services/EmailService";
+import { JwtService } from "@nestjs/jwt";
+import { KycService } from "../services/KycService";
 
 //TODO: need to fix the unit test later
```
Comment (brettvniekerk) [inline src/user/user.controller.spec.ts:12]: is this done?

File: src/user/user.service.ts [hunk 4]
```diff
@@ -86,45 +62,64 @@ export class UserService {
...
+            // Should not create user if email from token is different from email from public key
+            if (hashMessage(emailFromPublicKey) != newUser.email)
+                throw new Error("Email does not match");
+
+            user = await this.userRepository.findOneBy({
+                email: newUser.email
+            });
+
+            // Do nothing if user already exists and email is verified
+            if (user && user.emailVerified) return;
```
Comment (brettvniekerk) [inline src/user/user.service.ts:82]: Do we not want some kind of descriptive message like "User already exists"?

**Review by brettvniekerk — DISMISSED** [review]
Just a few optional touch ups, otherwise LGTM

**Review by JasminFarhi — APPROVED** [review]

**Comment by JasminFarhi** [comment]
LGTM, but not super familiar with back end so might be good to get another review.

File: src/exchange/exchange.service.ts (hunk 1 above)
Comment (xiangzhe0812) [reply src/exchange/exchange.service.ts:3]: good point, but not gonna do it on this ticket, its a bug refactor already.

File: src/otp/otp.controller.spec.ts (hunk 2 above)
Comment (xiangzhe0812) [reply src/otp/otp.controller.spec.ts:8]: good point, but not gonna do it on this ticket, its a bug refactor already

**Review by brettvniekerk — APPROVED** [review]

**Pull request merged**